### PR TITLE
[SMALLFIX] Allow user to list available validation tasks

### PR DESCRIPTION
**Pull request by bf8086**
This change added a new option `-l` to `validateEnv` command for user to list validation tasks to run for a given target and prefix. For example `alluxio validateEnv all -l` will list all tasks available to run for each master node and worker node.

### CODE REVIEW

**Comment by bf8086** [comment]
@aaudiber PTAL. thanks!

**Comment by aaudiber** [comment]
What do you think of simplifying the API to a single `listChecks` subcommand?

```bash
$ alluxio validateEnv listChecks
check1: description
check2: description
check3: description
```

If we list the checks in alphabetical order, users can scan the list to see which prefix will perform which checks.

**Comment by bf8086** [comment]
@aaudiber The command currently expect the first fixed position argument to be the `target` argument, so we will have to nest the `target` in another subcommand to add a new subcommand. An alternative approach is to make `validateEnv -l` equivalent to `validateEnv all -l` command so user will get a list of all commands they can run on master or worker nodes. What do you think?

**Comment by aaudiber** [comment]
We could update the help message to call it COMMAND instead of TARGET, similar to `bin/alluxio`
```bash
Usage: alluxio COMMAND [GENERIC_COMMAND_OPTIONS] [COMMAND_ARGS]
```

File: core/server/common/src/main/java/alluxio/cli/ValidateEnv.java [hunk 1]
```diff
@@ -80,54 +83,75 @@
...
+        "validate proxy web port is available",
         new PortAvailabilityValidationTask(ServiceType.PROXY_WEB, ALLUXIO_PROXY_CLASS),
         COMMON_TASKS);
 
     // security configuration validations
     registerTask("master.ufs.hdfs.security.kerberos",
+        "validate kerberos security configurations for masters",
         new SecureHdfsValidationTask("master"), MASTER_TASKS);
-    registerTask("worker.ufs.hdfs.security.kerberos",
+    registerTask("worker.ufs.hdfs.security.kerberos for workers",
```
Comment (aaudiber) [inline core/server/common/src/main/java/alluxio/cli/ValidateEnv.java]: "for workers" goes on the line below

**Comment by bf8086** [comment]
@aaudiber I made some changes per your suggestion. PTAL.

File: core/server/common/src/main/java/alluxio/cli/ValidateEnv.java (hunk 1 above)
Comment (bf8086) [reply core/server/common/src/main/java/alluxio/cli/ValidateEnv.java]: Good catch. Fixed.

**Pull request merged**